Remotion/Web/ClientScript/Scripts/SmartPage.ts

```diff
@@ -499,14 +503,27 @@ class SmartPage_Context
   public SetCacheDetectionFieldLoaded(): void
   {
     window.sessionStorage.setItem(this.GetPageToken(), SmartPage_CacheDetection[SmartPage_CacheDetection.HasLoaded]);
+    this.ClearOldPageTokenSessionStorageEntries();
   };
 
   // Marks the page as submitted.
   public SetCacheDetectionFieldSubmitted(): void
   {
     window.sessionStorage.setItem(this.GetPageToken(), SmartPage_CacheDetection[SmartPage_CacheDetection.HasSubmitted]);
+    this.ClearOldPageTokenSessionStorageEntries();
   };
 
+  private ClearOldPageTokenSessionStorageEntries(): void
+  {
+    const pageTokenEntries = Object.keys(window.sessionStorage).map((k: string) => +k).filter(t => !isNaN(t));
+    const pageTokenEntryCount = pageTokenEntries.length
+    if (pageTokenEntryCount > this._pageTokenDeletionThreshold)
+    {
+      const pageTokensToDelete = pageTokenEntries.sort().splice(0, pageTokenEntryCount - this._pageTokenDeletionKeepCount);
+      setTimeout(() => pageTokensToDelete.forEach(t => window.sessionStorage.removeItem(t.toString())), this._pageTokenDeletionDelay);
```

Comment:
is the timeout of > 0ms meant to actually delay the cleanup or simply to make it asynchronously. In the second case, it's enough to hardcode '0'.